src/replace-libraries-ngpackagr-builder.ts

```diff
@@ -0,0 +1,81 @@
+import {AngularJson} from './types/angular-json';
+import {cloneDeep} from 'lodash-es';
+import {AngularVersion} from './types/angular-version';
+import {angularVersionComparer} from './angular-version-comparer';
+import {ProjectConfiguration} from './types/project-configuration';
+import {ArchitectConfiguration} from './types/architect-configuration';
+
+export const preAngular10_1NgPackagrBuilder =
+  '@angular-devkit/build-ng-packagr:build';
+export const angular10_1AndUpNgPackagrBuilder =
+  '@angular-devkit/build-angular:ng-packagr';
+
+const angular10_1: AngularVersion = '10.1.x';
+
+/**
+ * Replace all libraries builder using ng-packagr with the correct builder given the Angular version
+ *
+ */
+export function replaceLibrariesNgPackagrBuilder(
+  angularVersion: AngularVersion,
+  angularJson: AngularJson
+): AngularJson {
+  const modifiedAngularJson = cloneDeep(angularJson);
+
+  const projectsAndtargetsWithNgPackagrBuilder = getNameOfProjectsAndTargetsUsingNgpackagr(
+    angularJson
+  );
+
+  for (const [
+    projectName,
+    targetName
+  ] of projectsAndtargetsWithNgPackagrBuilder) {
+    modifiedAngularJson.projects[projectName].architect[
+      targetName
+    ].builder = getCorrectNgPackgrBuilder(angularVersion);
+  }
```

Comment:
suggestion: Can we disable the `github/array-foreach` lint rule and use Array extensions instead?
```suggestion

  projectsAndtargetsWithNgPackagrBuilder.forEach(
    ([projectName, targetName]) => {
      modifiedAngularJson.projects[projectName].architect[
        targetName
      ].builder = getCorrectNgPackgrBuilder(angularVersion);
    }
  );
``